src/main/java/tlschannel/util/Util.java

```diff
@@ -31,13 +31,16 @@ public static String resultToString(SSLEngineResult result) {
   public static int getJavaMajorVersion() {
     String version = System.getProperty("java.version");
     if (version.startsWith("1.")) {
-      version = version.substring(2, 3);
-    } else {
-      int dot = version.indexOf(".");
-      if (dot != -1) {
-        version = version.substring(0, dot);
-      }
+      version = version.substring(2);
     }
-    return Integer.parseInt(version);
+    // Allow these formats:
```

Comment:
I'm curious what the reason is for not using `Runtime.Version.parse(System.getProperty("java.version")).major()`?  It's deprecated, but does the right thing for all but the first of these examples (which I'm not sure is valid).

Reply:
Runtime.version() was introduced in Java 9

Reply:
Ah, ok.  How about `Runtime.Version.parse(System.getProperty("java.version"))`?

Reply:
OK, I relent. Sorry for the noise.  I do appreciate the continued Java 8 support.